src/agent/configuration_parser/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.18)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_BUILD_TYPE Release)
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
```

Comment:
We should start using `RelWithDebInfo`.

Also, to my understanding `CMAKE_CXX_FLAGS_DEBUG` only sets flags for `Debug` build types, you can use instead `CMAKE_CXX_FLAGS`.